pennylane/optimize/rotoselect.py

```diff
@@ -105,25 +105,26 @@ def step(self, objective_fn, x, generators):
                 operators to be used in the circuit and optimized over
 
         Returns:
-            array: The new variable values :math:`x^{(t+1)}` as well as the new generators.
+            tuple: a tuple containing the new variable values :math:`x^{(t+1)}`, the objective
+                function output, as well as the new generators.
         """
         x_flat = np.fromiter(_flatten(x), dtype=float)
         # wrap the objective function so that it accepts the flattened parameter array
         objective_fn_flat = lambda x_flat, gen: objective_fn(unflatten(x_flat, x), generators=gen)
 
         try:
             assert len(x_flat) == len(generators)
-        except AssertionError:
+        except AssertionError as e:
             raise ValueError(
                 "Number of parameters {} must be equal to the number of generators.".format(x)
-            )
+            ) from e
 
         for d, _ in enumerate(x_flat):
             x_flat[d], generators[d] = self._find_optimal_generators(
                 objective_fn_flat, x_flat, generators, d
             )
-
-        return unflatten(x_flat, x), generators
+        x = unflatten(x_flat, x)
+        return x, generators
```

Comment:
I think it's worth also reverting this, as we can minimize required line changes in this PR